src/rpdk/core/jsonutils/resolver.py

```diff
@@ -131,16 +133,18 @@ def _get_multiple_lang_type(schema_type):
         return ResolvedType(ContainerType.MULTIPLE, schema_type)
 
     @staticmethod
-    def _get_primitive_lang_type(schema_type):
-        return ResolvedType(ContainerType.PRIMITIVE, schema_type)
+    def _get_primitive_lang_type(schema_type, property_schema):
+        return ResolvedType(
+            ContainerType.PRIMITIVE, schema_type, property_schema.get("format", DEFAULT)
```

Comment:
what if the format is not supported? this looks it can cause uncaught runtime exceptions. 
```json
{
    "type": "integer",
    "format": "int0"
}
```

Reply:
see aws-cloudformation/cloudformation-cli-java-plugin#328. Since language plugin would decide if a format is supported or not, was just going to default if not found.

```
            LOG.error(
                "Could not find specified format '%s' for type '%s'. "
                "Defaulting to '%s'",
                resolved_type.type_format,
                resolved_type.type,
                primitive_format,
            )
```

Reply:
yep, looked at both as one PR. maybe we should have a map of allowed formats on the cli side? so we can default it once and not rewrite logic on plugin side. lmk what u think

Reply:
i feel like that be cumbersome to maintain if we start adding a bunch of them to all of the different plugins. what if a certain type like long is supported in java [but not in python?](https://www.python.org/dev/peps/pep-0237/). I think that these lists should be provided by each plugin instead.